Purity: don't look for library dependencies (of the form `-lfoo') in
/lib and /usr/lib.  Likewise, when searching for included Makefiles,
don't look in /usr/include and friends.

Patch from Nixpkgs, by Eelco Dolstra.

diff -rc make-3.81-orig/src/read.c make-3.81/src/read.c
*** make-3.81-orig/src/read.c	2006-03-17 15:24:20.000000000 +0100
--- make-3.81/src/read.c	2007-05-24 17:16:31.000000000 +0200
***************
*** 99,107 ****
--- 99,109 ----
  #endif
      INCLUDEDIR,
  #ifndef _AMIGA
+ #if 0    
      "/usr/gnu/include",
      "/usr/local/include",
      "/usr/include",
+ #endif    
  #endif
      0
    };
diff -rc make-3.81-orig/src/remake.c make-3.81/src/remake.c
*** make-3.81-orig/src/remake.c	2006-03-20 03:36:37.000000000 +0100
--- make-3.81/src/remake.c	2007-05-24 17:06:54.000000000 +0200
***************
*** 1452,1460 ****
--- 1452,1462 ----
    static char *dirs[] =
      {
  #ifndef _AMIGA
+ #if 0
        "/lib",
        "/usr/lib",
  #endif
+ #endif
  #if defined(WINDOWS32) && !defined(LIBDIR)
  /*
   * This is completely up to the user at product install time. Just define
